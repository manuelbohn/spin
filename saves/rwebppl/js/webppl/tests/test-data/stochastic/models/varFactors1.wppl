var scalingWt = 0.1;

var foo = function(x) {
  var fn = function(os, acc, pf) {
    if (os.length == 0) {
      factor(-pf);
      return acc;
    } else {
      var nf = -scalingWt * uniformDraw(os);
      factor(nf - pf);
      return fn(os.slice(1), acc.concat(os[0]), nf);
    }
  };
  return fn(x, [], 0);
};

var model = function() {
  var x = uniformDraw([[10], [20, 10, 5]]);
  var e = foo(x);
  return e;
};
