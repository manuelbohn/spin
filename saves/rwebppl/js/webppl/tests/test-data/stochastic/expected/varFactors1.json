{
  "hist": {
    "[20,10,5]": 0.50,
    "[10]": 0.50
  }
}
